share/spice/bike_sharing/citi_bike_nyc/bike_sharing_citi_bike_nyc.js

```diff
@@ -0,0 +1,79 @@
+(function (env) {
+    "use strict";
+
+    env.ddg_spice_bike_sharing_citi_bike_nyc = function(api_result) {
+        if (!api_result || !api_result.stationBeanList) {
+            return Spice.failed('citibikenyc');
+        }
+
+        DDG.require('moment.js', function() {
+        DDG.require('maps', function() {
+            var script = $('[src*="/js/spice/bike_sharing/citi_bike_nyc/"]')[0];
+            var source = $(script).attr("src");
+            var query = source.match(/citi_bike_nyc\/([^\/]+)/)[1];
+
+            if (query) {
+                var references = {
+                    'brooklyn': L.latLng(40.645244, -73.9449975),
+                    'manhattan': L.latLng(40.7590615, -73.969231),
+                    'queens': L.latLng(40.651018, -73.871192)
+                };
+                var pointOfReference = references[query];
+            }
+
+            function normalize(item) {
+                if (item.testStation || item.statusKey != 1) {
+                    return null;
+                }
+                var subtitle = item.availableBikes + ' bike' + (item.availableBikes == 1 ? '' : 's') + ' | ' + item.availableDocks + ' dock' + (item.availableDocks == 1 ? '' : 's');
```

Comment:
We have some awesome helpers you can use to do this:

```js
var subtitle = [
                    item.availableBikes + DDG.pluralize(item.availableBikes, "bike"),
                    item.availableDocks + DDG.pluralize(item.availableDocks, "dock")
                ];
```

If you pass an array to `subtitle` we'll automatically add the separator for you. `DDG.Pluralize()` will handle pluralization as well :smile: